typescript error:  'undefined' is not assignable to parameter of type 'URLListener'.

### Summary

In version 1.0.0-rc8+ I get this type error.

``` 
node_modules/expo-router/src/link/linking.ts(91,36): error TS2345: Argument of type '(({ url }: { url: string; }) => void) | undefined' is not assignable to parameter of type 'URLListener'.
Type 'undefined' is not assignable to type 'URLListener'.
 ELIFECYCLE  Command failed with exit code 2.
```

### Minimal reproducible example

To recreate this you can run tsc on a project using the `"expo-router/ts-declarations",` type root.

I can provide a example project if that helps

--- a/packages/expo-router/CHANGELOG.md
+++ b/packages/expo-router/CHANGELOG.md
@@ -10,6 +10,8 @@
 
 ### 🐛 Bug fixes
 
+- Drop legacy `Linking.removeEventListener` method.
+
 ### 💡 Others
 
 - Make `react-native-gesture-handler` non-optional as Metro doesn't support optional dependencies correctly.

--- a/packages/expo-router/CHANGELOG.md
+++ b/packages/expo-router/CHANGELOG.md
@@ -11,3 +11,5 @@
 ### 🐛 Bug fixes
 
 ### 💡 Others
+
+- Make `react-native-gesture-handler` non-optional as Metro doesn't support optional dependencies correctly.

--- a/packages/expo-router/package.json
+++ b/packages/expo-router/package.json
@@ -53,9 +53,6 @@
     "react-native-screens": "*"
   },
   "peerDependenciesMeta": {
-    "react-native-gesture-handler": {
-      "optional": true
-    },
     "react-native-reanimated": {
       "optional": true
     },

--- a/packages/expo-router/src/link/linking.ts
+++ b/packages/expo-router/src/link/linking.ts
@@ -76,20 +76,11 @@ export function addEventListener(listener: (url: string) => void) {
   } else {
     callback = ({ url }: { url: string }) => listener(url);
   }
-  const subscription = Linking.addEventListener("url", callback) as
-    | { remove(): void }
-    | undefined;
-
-  // Storing this in a local variable stops Jest from complaining about import after teardown
-  const removeEventListener = Linking.removeEventListener?.bind(Linking);
+  const subscription = Linking.addEventListener("url", callback);
 
   return () => {
     // https://github.com/facebook/react-native/commit/6d1aca806cee86ad76de771ed3a1cc62982ebcd7
-    if (subscription?.remove) {
-      subscription.remove();
-    } else {
-      removeEventListener?.("url", callback);
-    }
+    subscription.remove?.();
   };
 }
 

